src/github.com/getlantern/flashlight/client/client.go

```diff
@@ -50,18 +50,31 @@ type Client struct {
 	l net.Listener
 }
 
-// ListenAndServe makes the client listen for HTTP connections.  onListeningFn
-// is a callback that gets invoked as soon as the server is accepting TCP
-// connections.
+// Addr returns the address at which the client is listening, blocking until the
+// given timeout for an address to become available.
+func Addr(timeout time.Duration) (interface{}, bool) {
+	return addr.Get(timeout)
+}
+
+func (c *Client) Addr(timeout time.Duration) (interface{}, bool) {
+	return Addr(timeout)
+}
+
+// ListenAndServe makes the client listen for HTTP connections at a random port
+// on localhost. onListeningFn is a callback that gets invoked as soon as the
+// server is accepting TCP connections.
 func (client *Client) ListenAndServe(onListeningFn func()) error {
+	log.Debug("About to listen")
 	var err error
 	var l net.Listener
 
-	if l, err = net.Listen("tcp", client.Addr); err != nil {
-		return fmt.Errorf("Client proxy was unable to listen at %s: %q", client.Addr, err)
+	if l, err = net.Listen("tcp", "localhost:0"); err != nil {
```

Comment:
Fixed - caller can now specify an address explicitly, or just let Lantern randomly assign one.  In desktop app, I explicitly request localhost:8787, in mobile we'll let it randomly assign.